R/derive_vars_aage.R

```diff
@@ -157,44 +221,98 @@ derive_agegr_fda <- function(dataset, age_var, new_var) {
 #' data(dm)
 #'
 #' dm %>%
-#'   derive_agegr_ema(AGE, AGEGR1) %>%
+#'   derive_agegr_ema(age_var = AGE, new_var = AGEGR1) %>%
 #'   select(SUBJID, AGE, AGEGR1)
 #'
 #' data.frame(AGE = 1:100) %>%
-#'   derive_agegr_ema(age_var = AGE, new_var = AGEGR1)
+#'   derive_agegr_ema(age_var = AGE, age_unit = "years", new_var = AGEGR1)
 #'
 #' data.frame(AGE = 1:20) %>%
-#'   derive_agegr_ema(age_var = AGE, new_var = AGEGR1, adults = FALSE)
-derive_agegr_ema <- function(dataset, age_var, new_var, adults = TRUE) {
+#'   derive_agegr_ema(age_var = AGE, age_unit = "years", new_var = AGEGR1, adults = FALSE)
+derive_agegr_ema <- function(dataset, age_var, new_var, age_unit = NULL, adults = TRUE) {
```

Comment:
This was discussed in #172 and there shouldn't be studies which have both adults and non-adult subjects. I've kept it in for now but this could be removed.

Reply:
I am not sure if this is true. I did a search in the [EU Clinical Trials Register](https://www.clinicaltrialsregister.eu/ctr-search/search?query=&age=adolescent&age=adult&age=children) and got a lot of studies with adult and non-adult subjects.

What is the benefit of the `adults` parameter?

Reply:
Sure if there can be both adults and non-adults it would be best to remove this. It doesn't need to be there so I will remove it and set the groups continuously for all ages. It will still work for pediatric studies but there will be no need to specify explicitly which type of study is required.